### PR TITLE
PROGRAMMES-5824 Tests contributionsService

### PR DESCRIPTION
**Pull request by tatitati**

### CODE REVIEW

File: tests/Service/ContributionsService/FindByContributionToProgrammeTest.php [hunk 1]
```diff
@@ -2,49 +2,55 @@
...
-
-        $this->mockRepository->expects($this->once())
-            ->method('findByContributionTo')
-            ->with([$dbId], 'programme', false, 5, 10)
-            ->willReturn($dbData);
-
-        $result = $this->service()->findByContributionToProgramme($programme, 5, 3);
-        $this->assertEquals($this->contributionsFromDbData($dbData), $result);
+        return [
+          'CASE: default' => [300, 0, []],
```
Comment (votemike) [inline tests/Service/ContributionsService/FindByContributionToProgrammeTest.php]: These array keys aren't overly clear to me. Could you expand on them a bit?

File: tests/Service/ContributionsService/FindByContributionToProgrammeTest.php [hunk 2]
```diff
@@ -2,49 +2,55 @@
...
-            ->method('findByContributionTo')
-            ->with([$dbId], 'programme', false, 5, 10)
-            ->willReturn([]);
+        $contributions = $this->service()->findByContributionToProgramme($this->createMock(Programme::class));
 
-        $result = $this->service()->findByContributionToProgramme($programme, 5, 3);
-        $this->assertEquals([], $result);
+        $this->assertCount(count($fakeDbContributions), $contributions);
+        $this->assertContainsOnly(Contribution::class, $contributions);
+        $this->assertEquals($expectedPids, $this->extractPids($contributions));
```
Comment (votemike) [inline tests/Service/ContributionsService/FindByContributionToProgrammeTest.php]: I'm not overly keen on asserting the return value of a method in this test class equals a whole array. Is there a way to do this that requires less data manipulation?

File: tests/Service/ContributionsService/FindByContributionToSegmentsTest.php [hunk 3]
```diff
@@ -2,61 +2,64 @@
...
+use BBC\ProgrammesPagesService\Domain\Entity\Contribution;
+use BBC\ProgrammesPagesService\Domain\Entity\Segment;
+
 class FindByContributionToSegmentsTest extends AbstractContributionsServiceTest
 {
-    public function testFindByContributionToSegmentsDefaultPagination()
+    /**
+     * @dataProvider paginationProvider
+     */
+    public function testPagination($expectedLimit, $expectedOffset, array $paginationParams)
```
Comment (votemike) [inline tests/Service/ContributionsService/FindByContributionToSegmentsTest.php]: Typehints

File: tests/Service/ContributionsService/AbstractContributionsServiceTest.php [hunk 4]
```diff
@@ -11,24 +13,26 @@ public function setUp()
...
-        $this->setUpMapper('ContributionMapper', 'contributionFromDbData');
+        $this->setUpMapper('ContributionMapper', function ($dbContribution) {
+            return $this->createConfiguredMock(Contribution::class,
+                ['getPid' => new Pid($dbContribution['pid'])]
+            );
+        });
     }
 
-    protected function contributionsFromDbData(array $entities)
+    protected function service()
```
[the remark this answers is not in the document]
Comment (tatitati) [reply tests/Service/ContributionsService/AbstractContributionsServiceTest.php]: Maybe getService().......the service itself is not being mocked. What is being mocked exactly are the dependencies of this service. But that apart is keeping the original funcionallity (which is good! jejeje).

File: tests/Service/ContributionsService/FindByContributionToProgrammeTest.php (hunk 2 above)
Comment (tatitati) [reply tests/Service/ContributionsService/FindByContributionToProgrammeTest.php]: I aggree, but not sure if is possible.....we can add multiple assets in rack for each item, can be more transparent, but it also creates more noise with so many asserts....

My idea is to keep it as such. Basically with this we are testing that after creating the entity models (produced really by the mapper but in the tests for us) none of our results are discarded, reordered, etc, so we can continue focusing on the output.
@BPScott any thought?

Comment (votemike) [reply tests/Service/ContributionsService/FindByContributionToProgrammeTest.php]: Maybe a foreach?

Comment (tatitati) [reply tests/Service/ContributionsService/FindByContributionToProgrammeTest.php]: Actually that is extractPids:
```php
    protected function extractPids(array $contributions): array
    {
        return array_map(
            function ($contribution) {
                return (string) $contribution->getPid();
            },
            $contributions
        );
    }
```

Comment (votemike) [reply tests/Service/ContributionsService/FindByContributionToProgrammeTest.php]: But that hides what is happening.

Comment (tatitati) [reply tests/Service/ContributionsService/FindByContributionToProgrammeTest.php]: maybe we can create something more generic:

```php
function extractProperty(string $property, array $items){
   return ......
}
```

At least this avoid to have extractPids, extractDbIds, ......

Comment (votemike) [reply tests/Service/ContributionsService/FindByContributionToProgrammeTest.php]: I don't like extracting anything as part of tests

**Pull request merged**